internal/resolve/catalog_test.go

```diff
@@ -589,3 +589,65 @@ func genPackage(pkg string) *declcfg.DeclarativeConfig {
 		Deprecations: []declcfg.Deprecation{packageDeprecation(pkg)},
 	}
 }
+
```

Comment:
Could you also add a positive case here? i.e. given multiple catalogs and valid label selector, one catalog is chosen.

Reply:
I had to add in support for labels to the catalog walker, but I think I got it.